examples/common/optim/lion.py

```diff
@@ -0,0 +1,129 @@
+from typing import Tuple, Optional, Callable
+
+import torch
+from torch.optim.optimizer import Optimizer
+import logging
+import math
+from composer.utils import dist
+
+log = logging.getLogger(__name__)
+
+# functions
+
+def exists(val):
+    return val is not None
+
+# update functions
+
+
+# class
+
+class DecoupledLionW(Optimizer):
+    metric_functions = {
+        'l2_norm/moment':
+            lambda param, optim_state, step_tensor: torch.linalg.vector_norm(optim_state['exp_avg']),
+        'l2_norm/param':
+            lambda param, optim_state, step_tensor: torch.linalg.vector_norm(param.data),
+        'l2_norm/update':
+            lambda param, optim_state, step_tensor: torch.linalg.vector_norm(step_tensor),
+        'l2_norm/grad':
+            lambda param, optim_state, step_tensor: torch.linalg.vector_norm(param.grad),
+        'cosine/update_grad':
+            lambda param, optim_state, step_tensor: torch.nn.functional.cosine_similarity(
+                param.grad.flatten(), step_tensor.flatten(), dim=0),
+        'cosine/moment_grad':
+            lambda param, optim_state, step_tensor: torch.nn.functional.cosine_similarity(
+                param.grad.flatten(), optim_state['exp_avg'].flatten(), dim=0),
+    }
+
+    def __init__(
+        self,
+        params,
+        lr: float = 1e-4,
+        betas: Tuple[float, float] = (0.9, 0.99),
+        weight_decay: float = 0.0,
+    ):
+        assert lr > 0.
+        assert all([0. <= beta <= 1. for beta in betas])
+        if weight_decay >= 1e-3:
+            log.warning(
+                f'You are using a high value of `weight_decay={weight_decay}` for the `DecoupledLionW` optimizer. Are you sure you want to do this? '
+                f'Your model\'s weights will be multiplied by {1.0 - weight_decay} on every step!')
+
+        defaults = dict(
+            lr = lr,
+            betas = betas,
+            weight_decay = weight_decay
+        )
+
+        super().__init__(params, defaults)
+        
+        for group in self.param_groups:
+            group['initial_lr'] = group['lr']
+
+    @staticmethod
+    def lionw(p, grad, exp_avg, lr, initial_lr, wd, beta1, beta2) -> None:
+        # stepweight decay
+        if wd != 0:
+            decay_factor = (lr / initial_lr) if initial_lr else 1.0
+            p.data.mul_(1 - decay_factor * wd)
+
+        # update is interpolation between gradient and momentum
+        update = exp_avg.clone().lerp_(grad, 1 - beta1).sign_()
```

Comment:
```suggestion
        update = exp_avg.lerp(grad, 1 - beta1).sign_()
```

I think this should create a new tensor without a need for a .clone()?